widgetry/src/widgets/plots.rs

```diff
@@ -0,0 +1,227 @@
+use std::collections::HashSet;
```

Comment:
Most of this was split out from `line_plot.rs`. Only real change is implementing `Axis` (previously `Yvalue`) for `Time`, and adding the second generic param to a few structs.